### ISSUE TITLE
Nullability of pattern variable on tuples causes unexpected compiler warnings

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#76297]
**Version Used**: https://github.com/dotnet/roslyn/commit/e3954f0cee4085436a590a14a01ec2f1b8a2571f

**Steps to Reproduce**:

```cs
     public static (object,object)? M1(object? a, object? b) => (a, b) is ({}, {}) t ? t : null; // warns (unexpected)
```    

**Expected Behavior**:

No warnings.

**Actual Behavior**:

The compiler warns about nullability mismatch.

`warning CS8619: Nullability of reference types in value of type '(object? a, object? b)' doesn't match target type '(object, object)?'`
